specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,209 @@
+Hit Test Kind
+===
+
+# Background
+The ability to ask WebView2 for information on a hit test is an important feature for 
+visually hosted apps. It can be used to support features related to mouse activity in 
+regions that are traditionally Non-Client areas. Features like Draggable Regions, 
+Resize, Minimize/Maximize functionality, and more.
+
+
+# Examples
+## Draggable regions (HWND)
+Draggable regions are marked regions on a webpage that will move the app window when the 
+user clicks and drags on them. It also opens up the system menu if right-clicked. This 
+API is designed to provide draggable regions support for apps that are directly hosting 
+the CoreWebView2 via visual hosting and for UI framework WebView2 controls that use visual 
+hosting, like the WinUI2 and WinUI3 WebView2 control.If your app uses windowed hosting, 
+draggable regions are supported by default, no need to follow this guide.
+
+NOTE: Your app may already be forwarding mouse messages to WebView2, if so the bulk 
+of the code below is likely already in your app. One thing you need to do specifically 
+for this feature is to add the check for w_param == HTCAPTION as a clause for capturing 
+the mouse and forwarding messages. It is necessary for the system menu to show. If you 
+choose not to include this you would have to handle the logic for showing the system menu 
+yourself.
+
+NOTE: This API is not currently supported for environments that use a CoreWindow like UWP
+
+#### Win32 C++
+```cpp
+LRESULT CALLBACK WndProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+....
+// The following code forwards mouse messages to the WebView2 - Start
+
+// handle the range of mouse messages & WM_NCRBUTTONDOWN
```

Comment:
I did some tests on this scenario. `WM_NCRBUTTON` only showed the system menu by default when it was on the actual window title bar. Returning `HTCPAPTION` inside the client area of the window did not cause the system menu to show.  It had to be shown manually.

Webview2 will show the system menu manually.

Reply:
Strange, because I'm looking at the DefWindowProc code, and on receipt of NCRBUTTONDOWN on HTCAPTION, it goes into a modal loop waiting for the RBUTTONUP. and then converts it to WM_CONTEXTMENU. The WM_CONTEXTMENU handler does another hit-test and looks for HTCAPTION. Maybe there's something weird that's sending us down an unusual code path that prevents this from working.

Reply:
Okay, I misread part of the DefWindowProc code. The problem is that the NCRBUTTONDOWN loop checks for HTCAPTION by doing hit-test calculations based on the window style and doesn't send WM_NCHITTEST to see if the app has custom caption areas. Sad.

Reply:
No follow up required.